BUILD.gn

```diff
@@ -0,0 +1,530 @@
+
+  # TODO: Add WebNotification for Linux
+  # WebNotification needs libgtk2ui and libnotify.
+
+  # TODO: Add nacl support.
+  if (enable_nacl) {
+    if (is_linux) {
+      sources += [
+        "runtime/browser/nacl_host/nacl_browser_delegate_impl.cc",
+        "runtime/browser/nacl_host/nacl_browser_delegate_impl.h",
+      ]
+    }
+    deps += [
+      "//components/nacl:nacl",
+      "//components/nacl:nacl_browser",
+      "//components/nacl:nacl_common",
+      "//components/nacl:nacl_helper",
+      "//components/nacl:nacl_linux",
+      "//components/nacl:nacl_renderer",
+      "//components/nacl/renderer/plugin:nacl_trusted_plugin",
+      "//native_client/src/trusted/service_runtime/linux:bootstrap",
+    ]
+  }
+
+  if (enable_plugins) {
+    deps += [
+      "//components/pdf/renderer:renderer",
+      "//ppapi/host:host",
+      "//ppapi/proxy:ipc",
+      "//ppapi/proxy:proxy",
+      "//ppapi/shared_impl",
+    ]
+    sources += [
+      "//chrome/renderer/pepper/pepper_flash_drm_renderer_host.cc",
+      "//chrome/renderer/pepper/pepper_flash_drm_renderer_host.h",
+      "//chrome/renderer/pepper/pepper_flash_font_file_host.cc",
+      "//chrome/renderer/pepper/pepper_flash_font_file_host.h",
+      "//chrome/renderer/pepper/pepper_flash_fullscreen_host.cc",
+      "//chrome/renderer/pepper/pepper_flash_fullscreen_host.h",
+      "//chrome/renderer/pepper/pepper_flash_menu_host.cc",
+      "//chrome/renderer/pepper/pepper_flash_menu_host.h",
+      "//chrome/renderer/pepper/pepper_flash_renderer_host.cc",
+      "//chrome/renderer/pepper/pepper_flash_renderer_host.h",
+      "runtime/browser/renderer_host/pepper/xwalk_browser_pepper_host_factory.cc",
+      "runtime/browser/renderer_host/pepper/xwalk_browser_pepper_host_factory.h",
+      "runtime/renderer/pepper/pepper_helper.cc",
+      "runtime/renderer/pepper/pepper_helper.h",
+      "runtime/renderer/pepper/pepper_uma_host.cc",
+      "runtime/renderer/pepper/pepper_uma_host.h",
+      "runtime/renderer/pepper/xwalk_renderer_pepper_host_factory.cc",
+      "runtime/renderer/pepper/xwalk_renderer_pepper_host_factory.h",
+    ]
+  }
+
+  if (disable_bundled_extensions) {
+    defines += [ "DISABLE_BUNDLED_EXTENSIONS" ]
+  }
+}
+
+action("generate_upstream_blink_version") {
+  upstream = "//xwalk/build/UPSTREAM.blink"
+  template = "//xwalk/runtime/browser/blink_upstream_version.h.in"
+  output_file = "$root_gen_dir/blink_upstream_version.h"
+
+  script = "//build/util/version.py"
+
+  args = [
+    "-f",
+    rebase_path(upstream, root_out_dir),
+    rebase_path(template, root_out_dir),
+    rebase_path(output_file, root_out_dir),
+  ]
+
+  outputs = [
+    "$output_file",
+  ]
+}
+
+#TODO: Remove this when there is no sign-comparation in xwalk code.
+config("no_sign_compare_warning") {
+  if (is_clang) {
+    cflags = [ "-Wno-sign-compare" ]
+  }
+}
+
+#TODO: Remove this when there is no unused const variables in xwalk code.
+config("no_unused_const_variable_warning") {
```

Comment:
Ditto.

In both cases, this is actually just a symptom of GN building with stricter compiler flags (`chromium_code` in gyp) by default. This is a bigger problem for Android (XWALK-6276), but fixing the errors that affect the Linux build should be really straight-forward.